File: src/main/java/examples/GreetingActor_3.java
package examples;

import java.util.concurrent.TimeUnit;

import akka.actor.AbstractActor;
import akka.actor.ActorSystem;
import akka.actor.Props;
import scala.concurrent.duration.Duration;

public class GreetingActor_3 extends AbstractActor {

    static class DoGreeting {
    }

    @Override
    public void preStart() throws Exception {
        scheduleNextGreeting();
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .match(DoGreeting.class, m -> {
                System.out.println("Hello!");
                scheduleNextGreeting();
            }).build();
    }

    private void scheduleNextGreeting() {
        ActorSystem system = context().system();
        system.scheduler().scheduleOnce(
            Duration.create(1, TimeUnit.SECONDS), self(), new DoGreeting(), system.dispatcher(), self());
    }

    public static void main(String[] args) {
        ActorSystem system = ActorSystem.create("MySystem");
        system.actorOf(Props.create(GreetingActor_3.class), "greeter");
    }
}

File: src/main/kotlin/examples/GreetingActor_3.kt
package examples

import akka.actor.ActorSystem
import akka.actor.Props
import akka.actor.UntypedActor
import scala.concurrent.duration.Duration
import java.util.concurrent.TimeUnit


class GreetingActor_3 : UntypedActor() {

    internal class DoGreeting

    override fun preStart() {
        scheduleNextGreeting()
    }

    override fun onReceive(msg: Any?) = when (msg) {
        is DoGreeting -> {
            println("Hello!")
            scheduleNextGreeting()
        }
        else -> {
            unhandled(msg)
        }
    }

    private fun scheduleNextGreeting() {
        val system = context().system()
        system.scheduler().scheduleOnce(
                Duration.create(1, TimeUnit.SECONDS), self(), DoGreeting(), system.dispatcher(), self())
    }
}

fun main(args: Array<String>) {
    val system = ActorSystem.create("MySystem")
    system.actorOf(Props.create(GreetingActor_3::class.java), "greeter")
}